modules/kubernetes-addons/crossplane/main.tf

```
@@ -43,134 +45,81 @@ resource "time_sleep" "wait_30_seconds" {
 }
 
 module "aws_provider_irsa" {
-  count                             = var.aws_provider.enable == true ? 1 : 0
+  count                             = local.aws_provider.enable == true ? 1 : 0
```

I think you get the gist - all of these should be updated similarly:
```suggestion
  count                             = try(local.aws_provider.enable, true) ? 1 : 0
```